src/bitcoin/wallet.rs

```diff
@@ -161,6 +161,36 @@ impl Wallet {
         }
     }
 
+    /// Wallet descriptors as specified in https://github.com/bitcoin/bitcoin/blob/master/doc/descriptors.md
+    pub fn descriptors(&self) -> Vec<String> {
+        let ext_priv_key = self.root_extended_private_key();
+        self.hd_paths()
+            .iter()
+            .map(|path| format!("wpkh({}{})", ext_priv_key, path))
+            .collect()
+    }
+
+    /// Some bitcoind rpc command requires the descriptor to be
+    /// suffixed with a checksum. For now we are going to ask bitcoind
+    /// to calculate the checksum for us.
+    pub async fn descriptors_with_checksums(&self) -> anyhow::Result<Vec<String>> {
+        let mut descriptors = Vec::new();
+        for descriptor in self.descriptors() {
+            let response = self
+                .bitcoind_client
+                .get_descriptor_info(&descriptor)
+                .await?;
+            let descriptor = format!("{}#{}", descriptor, response.checksum);
+            descriptors.push(descriptor);
+        }
+
+        Ok(descriptors)
+    }
+
+    fn hd_paths(&self) -> Vec<&str> {
```

Comment:
❓ How come there are two? Are these both used by bitcoind? Could/should we define a constant instead? It would help to clarify with a docstring.